src/controllers/controller.bar.js

```diff
@@ -294,8 +304,39 @@ module.exports = function(Chart) {
 			model.backgroundColor = custom.backgroundColor ? custom.backgroundColor : helpers.getValueAtIndexOrDefault(dataset.backgroundColor, index, rectangleElementOptions.backgroundColor);
 			model.borderColor = custom.borderColor ? custom.borderColor : helpers.getValueAtIndexOrDefault(dataset.borderColor, index, rectangleElementOptions.borderColor);
 			model.borderWidth = custom.borderWidth ? custom.borderWidth : helpers.getValueAtIndexOrDefault(dataset.borderWidth, index, rectangleElementOptions.borderWidth);
-		}
+		},
+
+		/**
+		 * @deprecated
+		 * @private
+		 */
+		calculateBarWidth: function(ruler) {
```

Comment:
Right, if they are overridden they won't be called anymore. I think the recommended way to achieve the old behaviour would be to override `updateElement` to set the properties as expect. I am fine to remove these since they won't do anything and overridding them won't work. We should wait to do anything until we see what users say